drivers/media/i2c/imx219.c

```diff
@@ -1127,6 +1182,82 @@ static int imx219_get_selection(struct v4l2_subdev *sd,
 	return -EINVAL;
 }
 
+static int imx219_set_selection(struct v4l2_subdev *sd,
+				struct v4l2_subdev_state *state,
+				struct v4l2_subdev_selection *sel)
+{
+	struct v4l2_rect max_bounds;
+	max_bounds.top = IMX219_PIXEL_ARRAY_TOP;
+	max_bounds.left = IMX219_PIXEL_ARRAY_LEFT;
+	max_bounds.width = IMX219_PIXEL_ARRAY_WIDTH;
+	max_bounds.height = IMX219_PIXEL_ARRAY_HEIGHT;
+	switch (sel->target) {
+	case V4L2_SEL_TGT_CROP: {
+		struct imx219 *imx219 = to_imx219(sd);
+
+		if (!v4l2_rect_enclosed(&sel->r, &max_bounds)) {
```

Comment:
Should you check that you're not streaming at this point?

Reply:
I dont see the check in S_FMT either, and I didnt spot it in `IMX274` driver too.